pkg/envinfo/envinfo.go

```diff
@@ -180,6 +190,13 @@ func (esi *EnvSpecificInfo) SetConfiguration(parameter string, value interface{}
 		case "push":
 			pushCommandValue := value.(EnvInfoPushCommand)
 			esi.componentSettings.PushCommand = &pushCommandValue
+		case "link":
+			linkValue := value.(EnvInfoLink)
+			if esi.componentSettings.Link != nil {
+				*esi.componentSettings.Link = append(*esi.componentSettings.Link, linkValue)
```

Comment:
shouldn't we check if there is a link with the same value present already to avoid duplicate linking?

Reply:
It happens at an earlier stage where we try to create the link using dynamic client:
```
$ odo link EtcdCluster/example
 ✗  servicebindingrequests.apps.openshift.io "node-todo-etcdcluster-example" already exists
```

Reply:
Replaced above message with below:
```sh
$ odo link EtcdCluster/example
 ✗  Component "node-todo" is already linked with the service "EtcdCluster/example"
```